### PR TITLE
feat: Implement delete deviceCommand

### LINKED ISSUE #651: Implement testcase for DELETE device command
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]
This feature request is for metadata

### Description [**REQUIRED**]
According to https://docs.edgexfoundry.org/2.2/design/adr/core/0021-Device-Profile-Changes/ & https://github.com/edgexfoundry/edgex-go/issues/3925
The DELETE device command implement tests would be added to metadata.

### Describe the solution you'd like
 If you have a solution in mind, please describe it.

### Describe alternatives you've considered
 Have you considered any alternative solutions or workarounds?

### PR DESCRIPTION
**Pull request by cherrycl**
Fix #651 
Signed-off-by: Cherry Wang <cherry@iotechsys.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-DeviceCommand.robot [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    And Create device profile ${deviceProfile}
+    When Delete deviceCommand by Name non-existent in Test-Profile-1
+    Then Should Return Status Code "404"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    [Teardown]  Delete device profile by name  ${profile_name}
+
+ErrProfileCommandDELETE003 - Delete deviceCommand which profile used by device
+    Given Set Test Variable  ${profile_name}  Test-Profile-2
+    Create A Device Sample With Associated Test-Device-Service And ${profile_name}
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-DeviceCommand.robot:45]:
```suggestion
    And Create A Device Sample With Associated Test-Device-Service And ${profile_name}
```

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-DeviceCommand.robot [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    Then Should Return Status Code "404"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+
+ErrProfileCommandDELETE002 - Delete deviceCommand by non-existent command name
+    # non-existent deviceCommand name
+    Given Set Test Variable  ${profile_name}  Test-Profile-1
+    And Generate A Device Profile Sample  ${profile_name}
+    And Create device profile ${deviceProfile}
+    When Delete deviceCommand by Name non-existent in Test-Profile-1
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-DeviceCommand.robot:37]:
```suggestion
    When Delete deviceCommand by Name non-existent in ${profile_name}
```

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-DeviceCommand.robot (hunk 1 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-DeviceCommand.robot]: Updated.

File: TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-DeviceCommand.robot (hunk 2 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/core-metadata/deviceprofile/DELETE-DeviceCommand.robot]: Updated

**Review by jumpingliu — APPROVED** [review]
LGTM

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**